entities/oracle_data.go

```diff
@@ -95,3 +98,54 @@ func (od *OracleData) ToProto() *oraclespb.OracleData {
 		BroadcastAt:    od.BroadcastAt.UnixNano(),
 	}
 }
+
+func (od OracleData) Cursor() *Cursor {
+	pks := make([]string, 0, len(od.PublicKeys))
+
+	for _, pk := range od.PublicKeys {
+		pks = append(pks, hex.EncodeToString(pk))
+	}
+
+	return NewCursor(OracleDataCursor{
+		VegaTime:   od.VegaTime,
+		PublicKeys: pks,
+	}.String())
+}
+
+func (od OracleData) ToProtoEdge(_ ...any) *v2.OracleDataEdge {
+	return &v2.OracleDataEdge{
+		Node:   od.ToProto(),
+		Cursor: od.Cursor().Encode(),
+	}
+}
+
+type OracleDataCursor struct {
+	VegaTime   time.Time `json:"vegaTime"`
+	PublicKeys []string  `json:"publicKeys"`
+}
+
+func (c OracleDataCursor) String() string {
+	bs, err := json.Marshal(c)
+	if err != nil {
+		builder := strings.Builder{}
+		builder.WriteString(fmt.Sprintf(`{"vegaTime":"%s","publicKeys":[`, c.VegaTime.Format(time.RFC3339Nano)))
+		for i, k := range c.PublicKeys {
+			if i > 0 {
+				builder.WriteString(`,`)
+			}
+			builder.WriteString(fmt.Sprintf(`"%s"`, k))
+		}
+		builder.WriteString(`]}`)
+		return builder.String()
```

Comment:
How come we have to do manual JSON marshalling here?

Reply:
It's just a fallback in case the marshalling fails because we definitely need a cursor and I'm being overly cautious

Reply:
hm, I feel if we can't marshal it it's probably a bug we should fix - also now we've got the possibility of different code for marshalling/unmarshalling; if you add a new field to the cursor struct you have to remember to add it here..

We marshal all these guys to JSON anyway when we put them in the DB so if it doesn't work something is very wrong!

Reply:
Okay, I will change them

Reply:
Thanks! (Reading around the code a bit I figured it's probably because you didn't want to have the possibility of returning an error here.. I think a panic() would probably be OK as it really shouldn't happen!)

Reply:
Thank you for the advice, changes applied here: https://github.com/vegaprotocol/data-node/pull/734/commits/e86748e4414ff58ea927998964e16a09edeef5e0